### ISSUE TITLE
Create option to crash the program on fatal exception

### ISSUE THREAD

**Issue by TheJJ** [SFTtech/openage#570]
Currently, all exceptions are caught. This is good, as one gets a shiny backtrace then.

When running in gdb, the program should terminate because of the exception, so one can directly debug from there.

This is rather easy: just rethrow the exception in the main catch clause. That way, the exception is thrown out of `main()` and the program (python.) gets `SIGABRT`ed, which is caught by gdb then.

tl;dr: add an option to rethrow an exception caught in the main function.

**Comment by TheJJ** [comment]
This is rather hard to do I think. Our exception translation mechanism inserts the cpp stack traces into the python stack trace (see `openage/cppinterface/exctranslate.pyx`).

So we would need to reraise the cpp error object through all the cython functions so it reaches the debugger at the root.

**Comment by TheJJ** [comment]
Another idea, which would be way simpler:
* The `openage::Error` constructor would run `asm("int $3;");` => gdb halts
* This instruction is only enabled by a launch option, e.g. `--trap-exceptions`

This would only catch openage exceptions though, which should be good for most debugging situations.

To catch `std::runtime_error`s as well, all the `try-catch` blocks in cython must be skipped, otherwise the stacktrace is clobbered. Damn you C++.

**Comment by Tomatower** [comment]
Fixed in #869
